### PR TITLE
[SCM-979] ScmLogger nonsense dropped

### PR DESCRIPTION
**Pull request by cstamas**
Using plain Slf4j logging. Anyone integrating
SCM should provide Slf4j implementation and
configure it as he likes.

PR targeting other PR https://github.com/apache/maven-scm/pull/141

---

https://issues.apache.org/jira/browse/SCM-979

### CODE REVIEW

**Comment by nielsbasjes** [comment]
Curious: Why SLF4J instead of the Log4j-api?

Essentially the same (logging facade) with 2 advantages:
- Better process (SLF4J has had a period of almost 2 years without any release).
- More features (also a formatted logger).

**Comment by michael-o** [comment]
> Curious: Why SLF4J instead of the Log4j2-api?

Our entire ecosystem is build around SLF4J

> Essentially the same (logging facade) with 2 advantages:
> 
>     * Better process (SLF4J has had a period of almost 2 years without any release).

What better process? Full of features most peole don't need? More releases aren't better. SFL4J is just a facade while Log4J2 offery EVERYTHING. There is not much you need to do if the facade is done.

>     * More features (also a formatted logger).

Which we don't need, we barely manage to move off string concat.

At the end, we prefer over simplily rather than busload of features.

Log4J2, even as an API, is a no go for me.

File: maven-scm-api/src/main/java/org/apache/maven/scm/command/AbstractCommand.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 public abstract class AbstractCommand
     implements Command
 {
-    private ScmLogger logger = new Slf4jScmLogger( getClass() );
+    protected Logger logger = LoggerFactory.getLogger( getClass() );
```
Comment (slawekjaranowski) [inline maven-scm-api/src/main/java/org/apache/maven/scm/command/AbstractCommand.java:38]: better is private and to have separate logger for each class, in this way logger name will be inconsistent with usage.

Comment (cstamas) [reply maven-scm-api/src/main/java/org/apache/maven/scm/command/AbstractCommand.java:38]: eh, the class hierarchy is really deep, so to make each class has own logger? Meg.
Also, this change above is on par with what we had before: subclasses used getLogger to get logger from parent.... so effect is same sans getter for logger.

**Review by michael-o — APPROVED** [review]

File: maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/add/HgAddConsumer.java [hunk 2]
```diff
@@ -54,17 +52,17 @@ public void doConsume( ScmFileStatus status, String trimmedLine )
             File tmpFile = new File( workingDir, trimmedLine );
             if ( !tmpFile.exists() )
             {
-                if ( getLogger().isWarnEnabled() )
+                if ( logger.isWarnEnabled() )
                 {
-                    getLogger().warn( "Not a file: " + tmpFile + ". Ignored" );
+                    logger.warn( "Not a file: " + tmpFile + ". Ignored" );
```
Comment (olamy) [inline maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/add/HgAddConsumer.java:57]: why not changing/simplify this as well by removing the if and `logger.warn( "Not a file: {}. Ignored", tmpFile );`
this apply to plenty of lines

File: maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/remove/HgRemoveConsumer.java [hunk 3]
```diff
@@ -55,17 +53,17 @@ public void doConsume( ScmFileStatus status, String trimmedLine )
             File tmpFile = new File( workingDir, trimmedLine );
             if ( !tmpFile.exists() )
             {
-                if ( getLogger().isWarnEnabled() )
+                if ( logger.isWarnEnabled() )
                 {
-                    getLogger().warn( "Not a file: " + tmpFile + ". Ignored" );
+                    logger.warn( "Not a file: " + tmpFile + ". Ignored" );
```
Comment (olamy) [inline maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/remove/HgRemoveConsumer.java:58]: could be simplify

File: maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/status/HgStatusConsumer.java [hunk 4]
```diff
@@ -52,24 +50,24 @@ public void doConsume( ScmFileStatus status, String trimmedLine )
         File tmpFile = new File( workingDir, trimmedLine );
         if ( !tmpFile.exists() )
         {
-            if ( getLogger().isInfoEnabled() )
+            if ( logger.isInfoEnabled() )
             {
-                getLogger().info( "Not a file: " + tmpFile + ". Ignoring" );
+                logger.info( "Not a file: " + tmpFile + ". Ignoring" );
```
Comment (olamy) [inline maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/status/HgStatusConsumer.java:55]: could be simplify

File: maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/add/HgAddConsumer.java (hunk 2 above)
Comment (michael-o) [reply maven-scm-providers/maven-scm-provider-hg/src/main/java/org/apache/maven/scm/provider/hg/command/add/HgAddConsumer.java:57]: This PR is a mechanical replacement. Placeholders are next, but are right of course.

**Comment by jira-importer** [comment]
Resolve #1206

**Pull request merged**